crates/bevy_render/src/render_graph/nodes/camera_node.rs

```diff
@@ -116,15 +116,20 @@ pub fn camera_node_system(
         staging_buffer
     };
 
-    let matrix_size = std::mem::size_of::<[[f32; 4]; 4]>();
-    let camera_matrix: [f32; 16] =
-        (camera.projection_matrix * global_transform.compute_matrix().inverse()).to_cols_array();
+    let camera_pos = &global_transform.translation;
+    let matrix_size = std::mem::size_of::<[[f32; 4]; 5]>();
+    let mut camera_matrix_array: Vec<f32> = (camera.projection_matrix
+        * global_transform.compute_matrix().inverse())
+    .to_cols_array()
+    .to_vec();
+    camera_matrix_array.extend_from_slice(&[camera_pos.x, camera_pos.y, camera_pos.z, 1.0]);
+    let camera_gpu_data = camera_matrix_array.as_slice();
 
     render_resource_context.write_mapped_buffer(
         staging_buffer,
         0..matrix_size as u64,
         &mut |data, _renderer| {
-            data[0..matrix_size].copy_from_slice(camera_matrix.as_bytes());
+            data[0..matrix_size].copy_from_slice(camera_gpu_data.as_bytes());
```

Comment:
Is having two `write_mapped_buffer()` calls slower? If not, it would improve clarity to split the writes.

Otherwise, can we have do `data[0..projection.byte_len()].copy_from_slice(projection.as_bytes())` followed by `data[projection.byte_len()..buffer_size].copy_from_slice(position.as_bytes())` without impacting performance to much?

Reply:
Leave a PERF note here and move on IMO. This isn't critical to figure out now, but we should leave bread crumbs.

In order to answer questions like this properly I'd want a rendering benchmark suite, which is well beyond the scope of this PR.

Reply:
The open question is because IMO the current code makes it easy to make errors, because of how it defines the matrix_size and buffer size separately as a Vec 4 elements longer than the actual matrix and then puts the 4 elements of the pos in there.

Maybe it's easier if I just rewrite it how I like it better and you can review that.

Reply:
Please look at https://github.com/bevyengine/bevy/pull/1554/commits/45d1137b78504054050c3f9bf701751c695e29d9 for what I mean.

Reply:
I'm happy enough with this until we get something like #1611 done.